### DISCUSSION TITLE
Disk space error

### DISCUSSION THREAD

**Discussion by jodelamo** [renovatebot/renovate#16154] [Request Help]
Hi!

I'm not really sure what's going on here, but I was looking into why we haven't been getting updates reliably in the past few days/weeks, and a colleague of mine showed me the Mend Renovate Dashboard, where I noticed we get the error below for each job:

```
# after a few thousand lines...
DEBUG: Passing disk-space error up(branch="renovate_ioredis-4.x")
ERROR: Disk space error - skipping
DEBUG: Repository result: disk-space, status: onboarded, enabled: true, onboarded: true
```

Our config:

```json
{
  "extends": ["config:base", ":dependencyDashboard", "group:linters", ":pinAllExceptPeerDependencies"],
  "prConcurrentLimit": 3,
  "commitMessagePrefix": "[Renovate] ",
  "timezone": "Europe/Stockholm",
  "branchPrefix": "renovate_",
  "ignorePaths": ["**/terraform/**"]
}
```

If we're running out of disk space, where is that happening? Is it after cloning the repository? After also installing dependencies?

Appreciate any help with this!

**Comment by rarkins** [comment]
> If we're running out of disk space, where is that happening? Is it after cloning the repository? After also installing dependencies?

It's not possible to answer these questions with the limited information you have provided.

I suggest you scroll up from the error you found and try to find where it "really" happened. e.g. if it's during an npm install then you know.

**Reply by jodelamo** [reply]
Sorry for the vague description. The error seems to happen **after** `npm install`:

```
DEBUG: Performing node_modules install(branch="renovate_ioredis-4.x")
DEBUG: Using node constraint "^16.0.0" from package.json(branch="renovate_ioredis-4.x")
DEBUG: Performing npm dedupe(branch="renovate_ioredis-4.x")
# ...
DEBUG: Passing disk-space error up(branch="renovate_ioredis-4.x")
ERROR: Disk space error - skipping
DEBUG: Repository result: disk-space, status: onboarded, enabled: true, onboarded: true
```

I checked our repository size post install/build, and it's quite large coming in at 11 GB, so I'm assuming that could be a culprit.

A recent job ID: `717923817`

**Reply by PhilipAbed** [reply]
i'm not sure that will help much but :

- Does this happen every time u run? 
- is the npm install command failing or successful?
- How many repositories are you running on, check how much space Cloning those repos take
- How much does npm install take space
- are you running on your own docker machine or where?

**Reply by rarkins** [reply]
Unfortunately this repo is too large and times out in the app. You will need to explore self-hosting as at the moment our default machines cannot finish the repo in time.

**Reply by jodelamo** [reply]
Thank you. We're looking into ways of reducing the size.

Thanks for your time!
